service/storage_proxy.cc

```diff
@@ -5161,9 +5161,10 @@ storage_proxy::query_singular(lw_shared_ptr<query::read_command> cmd,
 
     try {
         auto timeout = query_options.timeout(*this);
-        auto handle_completion = [&used_replicas, tmptr] (std::pair<::shared_ptr<abstract_read_executor>, dht::token_range>& executor_and_token_range) {
+        // `erm` capture below protects `&tm`
+        auto handle_completion = [&used_replicas, &tm, erm] (std::pair<::shared_ptr<abstract_read_executor>, dht::token_range>& executor_and_token_range) {
```

Comment:
nit: why is capturing `erm` needed?
handle_completion is synchronous, and even if not, the surrounding function is a coroutine so it would keep erm alive.

Reply:
Right, probably not needed, will remove after verification.